statistics/handle/handle.go

```diff
@@ -251,28 +267,47 @@ func (h *Handle) updateStatsCache(newCache statsCache) {
 	h.statsCache.Lock()
 	oldCache := h.statsCache.Load().(statsCache)
 	if oldCache.version <= newCache.version {
+		h.statsCache.memTracker.Consume(newCache.memUsage - oldCache.memUsage)
 		h.statsCache.Store(newCache)
 	}
 	h.statsCache.Unlock()
 }
 
 func (sc statsCache) copy() statsCache {
-	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)), version: sc.version}
+	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)),
+		version:  sc.version,
+		memUsage: sc.memUsage}
 	for k, v := range sc.tables {
 		newCache.tables[k] = v
 	}
 	return newCache
 }
 
+func (sc statsCache) initMemoryUsage() {
+	sum := int64(0)
+	for _, tb := range sc.tables {
+		sum += tb.MemoryUsage()
+	}
+	sc.memUsage = sum
+	return
+}
+
 // update updates the statistics table cache using copy on write.
 func (sc statsCache) update(tables []*statistics.Table, deletedIDs []int64, newVersion uint64) statsCache {
 	newCache := sc.copy()
 	newCache.version = newVersion
 	for _, tbl := range tables {
 		id := tbl.PhysicalID
+		if ptbl, ok := newCache.tables[id]; ok {
+			newCache.memUsage -= ptbl.MemoryUsage()
```

Comment:
why minus here?

Reply:
if  newCache.tables[id] not nil, it means  remove old `tables[id]` and place new  `tables[id]`.so we should minus old tables[id].memUsage first.
if newCache.tables[id] is nil , we only need to put new table in `tables[id]`.